op-geth-proxy/geth-proxy.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+
+	"github.com/peterbourgon/ff/v3"
+)
+
+// Environment variables beginning with this prefix can be used to instantiate command line flags
+const ENV_PREFIX = "SEQUENCER_PROXY"
+
+var (
+	fs            = flag.NewFlagSet("proxy", flag.ContinueOnError)
+	fromAddr      = fs.String("listen-addr", "127.0.0.1:9090", "proxy's listening adress")
+	sequencerAddr = fs.String("seq-addr", "http://127.0.0.1:50000", "address of espresso sequencer")
+	gethAddr      = fs.String("geth-addr", "http://127.0.0.1:8545", "address of the op-geth node")
+	vm_id         = fs.Int("vm-id", 1, "VM ID of the OP rollup instance")
+)
+
+type Transaction struct {
+	Vm      int   `json:"vm"`
+	Payload []int `json:"payload"`
+}
+
+type rpcMessage struct {
+	Params json.RawMessage `json:"params,omitempty"`
+	Method string          `json:"method,omitempty"`
+}
+
+func ForwardToSequencer(message rpcMessage) {
+	// json.RawMessage is a []byte array, which is marshalled
+	// As a base64-encoded string. Our sequencer API expects a JSON array.
+	payload := make([]int, len(message.Params))
+	for i := range payload {
+		payload[i] = int(message.Params[i])
+	}
+
+	// Construct a transaction and send it to the sequencer
+	txn := Transaction{
+		Vm:      *vm_id,
+		Payload: payload,
+	}
+	marshalled, err := json.Marshal(txn)
+	if err != nil {
+		panic(err)
+	}
+	request, err := http.NewRequest("POST", *sequencerAddr+"/submit/submit", bytes.NewBuffer(marshalled))
```

Comment:
> `*sequencerAddr`

Is this....dereferencing the global variable which defines the sequencerAddr command line flag? Which at runtime somehow gets populated with the value of the flag? Is that how Go does command line parsing? Cuz....ew

Reply:
Haha yeah it's not pretty, perhaps there is a more idiomatic way of doing this